spotify_player/src/client/mod.rs

```diff
@@ -1282,6 +1300,37 @@ impl Client {
         Ok(())
     }
 
+    pub async fn create_new_playlist(
+        &self,
+        state: &SharedState,
+        user_id: UserId<'static>,
+        playlist_name: &str,
+        public: bool,
+        collab: bool,
+        desc: &str,
+    ) -> Result<PlaylistId> {
+        let resp = self
+            .spotify
+            .user_playlist_create(
+                user_id.to_owned(),
+                playlist_name,
+                Some(public),
+                Some(collab),
+                Some(desc),
+            )
+            .await?;
+        state.data.write().user_data.playlists.insert(
+            0,
+            Playlist {
+                id: resp.id.to_owned(),
+                collaborative: collab,
+                name: playlist_name.to_string(),
+                owner: ("".into(), user_id),
+            },
```

Comment:
should use `resp.into()` instead. `state::Playlist` does implement the `From<rspotify_model::FullPlaylist>` trait.